enumerables.rb

```diff
@@ -0,0 +1,123 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    n_array = []
+    while i < to_a.length
+      x = to_a[i]
+      n_array[i] = yield(x)
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    n_array = []
+    while i < to_a.length
+      x = to_a[i]
+      n_array[i] = yield(x, i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = self
+    n_array = []
+    arr.my_each { |x| n_array << x if yield x }
+    n_array
+  end
+
+  def my_all?(arg = nil)
+    arr = self
+    y = true
+    if block_given?
+      arr.my_each { |x| return false if yield(x) == false }
+      return true
+    elsif !block_given? && arg.nil?
+      return true
+    elsif arg.is_a?(Class)
+      arr.my_each { |x| return false unless x.is_a?(arg) }
+    elsif arg.is_a?(Regexp)
+      arr.my_each { |x| return false unless arg.match(x) }
+    else
+      arr.my_each { |x| return false if x != arg }
+    end
+    y
+  end
+
+  def my_any?(arg = nil)
+    arr = self
+    if block_given?
+      arr.my_each { |x| return true if yield x }
+    elsif !block_given? && arg.nil?
+      arr.my_each { return true unless arr.nil? }
+    elsif arg.is_a?(Class)
+      arr.my_each { |x| return true if x.is_a?(arg) }
+    elsif arg.is_a?(Regexp)
+      arr.my_each { |x| return true if arg.match(x) }
+    else
+      arr.my_each { |x| return true if x == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    arr = self
+    return !arr.my_any?(arg) unless block_given?
+
+    arr.my_each { |x| return false if yield x }
+    true
+  end
+
+  def my_count(arg = nil)
+    arr = self
+    y = 0
+    if block_given?
+      arr.my_each { |x| y += 1 if yield x }
+    elsif !block_given? && arg.nil?
+      y = arr.length
+    else
+      y = arr.my_select { |x| x == arg }.length
+    end
+    y
+  end
+
+  def my_map(arg = nil)
+    index = 0
+    n_array = []
+    while index < to_a.length
+      n_array[index] = arg.nil? ? yield(to_a[index]) : !yield(to_a[index])
+      index += 1
+    end
+    n_array
+  end
```

Comment:
- [ ] When no block is given, `#my_map` should return an enumerator.

```ruby
array = Array.new(100) { rand(0...9) }
p array.my_map => error
```
*  Yours currently returns a` LocalJumpError`:`no block given (yield)`